### PR TITLE
Update to latest Rust

### PR DESCRIPTION
**Pull request by rsaarelm**

### CODE REVIEW

File: src/map.rs [hunk 1]
```diff
@@ -173,18 +173,18 @@ impl<K, V, C> Default for TreeMap<K, V, C> where C: Compare<K> + Default {
     fn default() -> TreeMap<K, V, C> { TreeMap::with_comparator(Default::default()) }
 }
 
-impl<K, V, C, Q: ?Sized> ops::Index<Q> for TreeMap<K, V, C> where C: Compare<K> + Compare<Q, K> {
+impl<K, V, C, Q: Sized> ops::Index<Q> for TreeMap<K, V, C> where C: Compare<K> + Compare<Q, K> {
```
Comment (apasel422) [inline src/map.rs]: This should be

``` rust
impl<'a, K, V, C, Q: Sized> ops::Index<&'a Q> for TreeMap<K, V, C> where C: Compare<K> + Compare<Q, K> {
```

File: src/map.rs [hunk 2]
```diff
@@ -173,18 +173,18 @@ impl<K, V, C> Default for TreeMap<K, V, C> where C: Compare<K> + Default {
...
     #[inline]
-    fn index(&self, i: &Q) -> &V {
-        self.get(i).expect("no entry found for key")
+    fn index(&self, i: Q) -> &V {
+        self.get(&i).expect("no entry found for key")
     }
 }
 
-impl<K, V, C, Q: ?Sized> ops::IndexMut<Q> for TreeMap<K, V, C> where C: Compare<K> + Compare<Q, K> {
+impl<K, V, C, Q: Sized> ops::IndexMut<Q> for TreeMap<K, V, C> where C: Compare<K> + Compare<Q, K> {
```
Comment (apasel422) [inline src/map.rs]: Same here.

**Comment by rsaarelm** [comment]
I tried doing that, but now some test code won't compile. I don't really understand this bit well enough to get it right by myself, sorry.

**Comment by Gankra** [comment]
@rsaarelm Thanks for getting the ball rolling on this!

**Pull request closed without merging**